gobblin-utility/src/main/java/org/apache/gobblin/util/filesystem/ModTimeDataFileVersion.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.util.filesystem;
+
+import java.io.IOException;
+import java.util.Set;
+
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import com.google.common.collect.ImmutableSet;
+import com.typesafe.config.Config;
+
+import lombok.Data;
+
+
+/**
+ * An implementation of {@link DataFileVersion} that uses modtime as the file version.
+ */
+@Data
+public class ModTimeDataFileVersion implements DataFileVersion<Long> {
```

Comment:
When do we use this ModTimeDataFileVersion? I remember in the design document (https://docs.google.com/document/d/1VhCHE2xfmwf4WTfle_wFhz-1U3TttIWDMU_YL0jkUbE/edit), we will have 'version string' written by Purger, but at the same time we will also have a 'version timestamp' that is equivalent to modTime in order to detect the situation where a file is modified without updating the version. For both cases, we need both versions to be set in the XAttr, but here the setVersion returns false, did I miss something?

Reply:
Modtime is a sensible default for versioning strategy, and lets distcp work on versions comparison instead of modtime comparison. If no alternate strategy is specified, distcp just falls back to what it is doing now (because the default version is just the modtime), but if an alternate is specified, it will respect it.